src/sentry/monitors/logic/mark_ok.py

```diff
@@ -33,17 +21,44 @@ def mark_ok(checkin: MonitorCheckIn, ts: datetime):
         "next_checkin": next_checkin,
         "next_checkin_latest": next_checkin_latest,
     }
-    if checkin.status == CheckInStatus.OK:
-        if monitor_env.monitor.status != ObjectStatus.DISABLED:
-            params["status"] = MonitorStatus.OK
-        # in the future this will auto-resolve associated issues
-        if monitor_env.status != MonitorStatus.OK:
-            params["last_state_change"] = ts
-            # resolve any associated incidents
-            incidents = MonitorIncident.objects.filter(
-                monitor_environment=monitor_env, grouphash=monitor_env.incident_grouphash
+
+    if (
+        monitor_env.monitor.status != ObjectStatus.DISABLED
+        and monitor_env.status != MonitorStatus.OK
+    ):
+        params["status"] = MonitorStatus.OK
+        recovery_threshold = monitor_env.monitor.config.get("recovery_threshold", 0)
+
+        # Run incident logic if recovery threshold is set
+        if recovery_threshold:
+            # Safety check to prevent db lockup on check-in lookup
+            assert recovery_threshold is not None
+
+            # Check if our incident is recovering
+            previous_checkins = MonitorCheckIn.objects.filter(
+                monitor_environment=monitor_env
+            ).order_by("-date_added")[:recovery_threshold]
```

Comment:
This does make me think @rjo100 we **NEED** a maximum value here. Do we have one?

Reply:
i can put a max in a diff PR and then we can reference that value and do a `min`

Reply:
Yeah, we definitely need to have it. Otherwise someone is going to slap a huge number in here and break our whole system.

Reply:
https://github.com/getsentry/sentry/pull/57956